### DISCUSSION TITLE
[Feature] Provide working configuration with HTTPS (caddy)

### DISCUSSION THREAD

**Discussion by rekcodocker** [immich-app/immich#9332] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I want to use HTTPS because several browsers don't even support unsecured connections anymore.
When I google how to set it up, it seems that Caddy is a good candidate. 
However, I find only three types of articles:
1. "I am trying to get Caddy to work, it does not work, here is my configuration"
2. "It is easy, just use Caddy as a reverse proxy" - no further explanation on how to do that
3. Very very specific examples which assume that everybody is using very specific hosters and domain-services - the working example depends on API keys, provider-specific settings and screens etc. 

For example. 
Nextcloud All In One comes with a reverse proxy built in.  
You only need to do 2 things:
1. forward your 443 port to the machine which runs the Nextcloud containers.
2. input "what is the domain name of your server"

It would be nice if it would be that easy.

A working guide would also be nice.  Not "Yeah just connect Caddy as a reverse proxy and bob's your uncle" but a real working example which you can copy and paste and it actually works.  And does not depend on Cloudflare because those examples are not portable.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by alextran1502** [comment]
Hello, you can find the example on the documentation site here https://immich.app/docs/administration/reverse-proxy#caddy-example-config

**Reply by rekcodocker** [reply]
Thanks. That is a very good example of what I mean.
The example is literally 3 lines of code.  There is no explanation of:

- where to put this code
- what it means (in case your configuration is different)
- where to input your domain name
- what kind of Caddy installation you need (apparently it can be a webserver but also a reverse proxy). 
- There may or may not be a 'caddy network' which I assume is some network in Docker. Do I connect the Caddy network to Immich, and if so, how?

It is 3 lines - it is not a working example. These 3 lines only work when you insert them in a specific place in a specific file in a specific setup of Caddy - AND get some specific settings like host / domainname etcetera right - and there is not a hint on where to start.

**Reply by alextran1502** [reply]
We provide you the example configs for those reverse proxy engines. If the users want to look into implement one, the info should be found on those engine's website. It doesn't make sense to put a tutorial guide for each engine on Immich docs because it is not what Immich is for

**Reply by rekcodocker** [reply]
True but it is 2024. At some point security has become part of every application. 

To use Immich as a gallery outside the local network, this becomes an essential part.  My mobile phone refuses to even connect to an unsecured connection.

There does not need to be a tutorial on every type of reverse proxy.
Example, example:  Nextcloud-all-in-one just comes with a working reverse proxy. It doesn't have different options and it doesn't need them; it just uses one and it works out of the box.  The security is a component of the package, not something that every user bolts on differently.

**Reply by bo0tzz** [reply]
Caddy does https by default, there's no need for us to include it in the example config.

**Reply by rekcodocker** [reply]
Ok so I just do "docker run caddy"  and it works?

Here is my frustration: You guys say 'we're not going to explain because it's simple' .
But it's not as simple as 'just run this command and it works'. 

You are writing from the position where you know this already. I don't.  
I don't know what caddy is or how to configure it.
I just know that some sofware can bundle that knowledge in  docker compose file so it becomes a component and it just works. 

When you google "immich docker caddy" you don't find someone saying "copy this and it works". You find people with problems and non-working configurations.

**Reply by bo0tzz** [reply]
The reason we're not including all these details about how to set up these external tools is because there are plenty of good documentation and guides out there already, and we (reasonably) expect people to be able to find those. We don't have a guide specific to caddy for the same reason we don't have a guide for installing Docker, or for setting up Authentik, namely that we would just be spending all our time duplicating information that already exists elsewhere.

**Reply by bo0tzz** [reply]
> When you google "immich docker caddy" you don't find someone saying "copy this and it works". You find people with problems and non-working configurations.

Don't limit your searches to just Immich. If you search "docker caddy" or "caddy guide" there is so much good information out there. If you then run into issues applying those to immich, you can search "immich caddy" and find the solutions that people before you have already come up with.

**Reply by alextran1502** [reply]
I want to chime in a bit on this: exposing your instance to the internet requires a bit more knowledge about networking and understanding the implication/risk in the decision; copying/pasting the config alone and not understanding them will inevitably put your instance and your data at risk. So a little bit of a learning curve will help along the way :)
